### PR TITLE
Changed source site

### PR DESCRIPTION
**Pull request by pinlast**
Changed the main site (useragentstring.com ) where information was taken to developers.whatismybrowser.com, to get the newer information

### CODE REVIEW

File: .pytest_cache/v/cache/nodeids [hunk 1]
```diff
@@ -0,0 +1,38 @@
+[
```
Comment (hellysmile) [inline .pytest_cache/v/cache/nodeids:1]: add this to .gitignore

File: fake_useragent/utils.py [hunk 2]
```diff
@@ -113,37 +114,117 @@ def get_browsers(verify_ssl=True):
     return list(zip(browsers, browsers_statistics))
 
 
+def get_version(browser, info):
+    try:
+        temp = re.search(re.escape(browser.capitalize()) + r'/(\d+)',info)#serach smth like "Chrome/60"
+        return int(temp.group(0)[len(browser)+1:])#getting only version digits
+    except:
```
Comment (hellysmile) [inline fake_useragent/utils.py]: blind except is dangerous

File: fake_useragent/utils.py [hunk 3]
```diff
@@ -113,37 +114,117 @@ def get_browsers(verify_ssl=True):
...
-        settings.BROWSER_BASE_PAGE.format(browser=quote_plus(browser)),
-        verify_ssl=verify_ssl,
-    )
-    html = html.decode('iso-8859-1')
-    html = html.split('<div id=\'liste\'>')[1]
-    html = html.split('</div>')[0]
 
-    pattern = r'\?id=\d+\'>(.+?)</a'
-    browsers_iter = re.finditer(pattern, html, re.UNICODE)
+        print(settings.BROWSER_BASE_PAGE + ' settings.BROWSER_BASE_PAGE')
```
Comment (hellysmile) [inline fake_useragent/utils.py]: remove print and use str.format

File: fake_useragent/utils.py [hunk 4]
```diff
@@ -113,37 +114,117 @@ def get_browsers(verify_ssl=True):
...
+
+        useragents_list = []
+
+        #Because ie fucking special and we must handel it another way
+        if browser == 'internet-explorer':
+            return get_browser_versions_ie(browser)
+
+        for page in range(1,4):#getting info from first 3 pages
```
Comment (hellysmile) [inline fake_useragent/utils.py]: do flake8 checks please

File: fake_useragent/utils.py [hunk 5]
```diff
@@ -113,37 +114,117 @@ def get_browsers(verify_ssl=True):
...
+            html = get(
+                    "https://developers.whatismybrowser.com/useragents/explore/software_name/{}/{}?order_by=-times_seen".format(browser,page),
```
Comment (hellysmile) [inline fake_useragent/utils.py]: use named formatting please

File: fake_useragent/utils.py [hunk 6]
```diff
@@ -113,37 +114,117 @@ def get_browsers(verify_ssl=True):
...
+                    )
+
+
+            html = html.decode('utf-8')
+
+            #getting td and than tr tags
+            table = re.findall(r'<table class="table table-striped table-hover table-bordered table-useragents">(.*?)</table>', html, re.DOTALL)[0]
+            matchObj = re.findall(r'<tr>(.*?)</tr>', table, re.M|re.I|re.S)
+
+            for obj in matchObj[1:]:
```
Comment (hellysmile) [inline fake_useragent/utils.py]: CamelCase is not allowed in Python in mostly cases

File: fake_useragent/utils.py [hunk 7]
```diff
@@ -113,37 +114,117 @@ def get_browsers(verify_ssl=True):
...
-        if 'more' in browser.group(1).lower():
-            continue
 
-        browsers.append(browser.group(1))
+        return useragents_list[len(useragents_list):len(useragents_list)-50:-1]
 
-        if len(browsers) == settings.BROWSERS_COUNT_LIMIT:
+
+def get_version_ie(info):
+    if info[1].isdigit() == False:
```
Comment (hellysmile) [inline fake_useragent/utils.py]: comparing to False is not needed

File: fake_useragent/utils.py [hunk 8]
```diff
@@ -254,3 +335,5 @@ def load_cached(path, use_cache_server=True, verify_ssl=True):
 
 from fake_useragent import settings  # noqa # isort:skip
 from fake_useragent.errors import FakeUserAgentError  # noqa # isort:skip
+
+print(get_browser_versions('chrome'))
```
Comment (hellysmile) [inline fake_useragent/utils.py]: do not leave prints please

**Comment by andrewkcarter** [comment]
Bump. This is requested.

**Comment by andrewkcarter** [comment]
@hellysmile could I convince you to cut a new release with this update? As is, the library is favoring user agents from 2013/14 and several prominent web sites are redirecting to "Upgrade your browser " pages.

**Comment by andrewkcarter** [comment]
@pinlast have you pushed your fork up to pypi yet? Would love to use it.

**Comment by PeteFein** [comment]
Is this project still alive?

**Comment by andrewkcarter** [comment]
Appears not.

**Comment by adrianmeraz** [comment]
So I take it we're out of options here? The useragentstring.com source site doesn't even work any more, and the data used in this package is super stale.

**Comment by melroy89** [comment]
~~I will make a fork: https://github.com/danger89/fake-useragent, maybe add `developers.whatismybrowser.com` as a fallback site?~~

In the `master` branch the  https://useragentstring.com as source does work again. I might consider changing the source, only if the source is out-dated or wrong.

Update the package: `pip install --upgrade fake-useragent`

**Comment by melroy89** [comment]
Although the latest release is working again. We still want to move to a different source with more up to date useragent strings. See also: https://github.com/fake-useragent/fake-useragent/issues/109

For now I close this PR, because the source code is deviated too much in the meanwhile. A new PR is required.

**Pull request closed without merging**